### PR TITLE
More pylint fixes on the test code

### PR DESCRIPTION
**Pull request by marcospri**
Opened a second PR to make it more manageable, again only test code.

One commit per issue, @jon-betts, let me know if you rather me split any of those to its own PR.

### CODE REVIEW

File: tests/common/factories/authz_code.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 def generate_code(_=None):
     """Simulate the way oauthlib generates authz codes."""
-    chars = "abcdefghijklmnopqrstuvwxyz" "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "012a456789"
+    chars = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ" + "012a456789"
```
Comment (jon-betts) [inline tests/common/factories/authz_code.py:15]:
```suggestion
    chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ012a456789"
```

File: tests/h/activity/query_test.py [hunk 2]
```diff
@@ -175,20 +175,16 @@ def test_preserves_other_query_terms_for_user_search(
...
 
-    def test_does_nothing_if_not_on_search_page(self, pyramid_request, unparse):
+    def test_doesnt_raise_if_not_on_search_page(self, pyramid_request, unparse):
         pyramid_request.matched_route.name = "group_read"
         query = MultiDict({"group": "abcd1234"})
 
-        result = check_url(pyramid_request, query, unparse=unparse)
-
-        assert result is None
+        check_url(pyramid_request, query, unparse=unparse)
```
Comment (jon-betts) [inline tests/h/activity/query_test.py:187]:
```suggestion
        result = check_url(pyramid_request, query, unparse=unparse)
        
        assert not result
```

**Review by jon-betts — CHANGES_REQUESTED** [review]
The vast majority of this looks fine. I've sprinkled a few suggestions here and there, but there are a couple of places where I think we've changed the meaning of the test. 

I can't really tell if it's on purpose or not here.

I don't mind larger PRs for this. Honestly I wouldn't worry about breaking them into tiny commits either. Just pick a small set of files and fix everything in them.

File: tests/h/tasks/mailer_test.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def test_send_creates_email_message_with_html_body(
 ):
     celery.request = pyramid_request
 
-    mailer.send(
+    mailer.send(  # pylint:disable=no-value-for-parameter #: (binded celery task)
```
[the remark this answers is not in the document]
Comment (marcospri) [reply tests/h/tasks/mailer_test.py:35]: Yeah, the parameter is `bind=True` but I don't think that sentence of mine is English

File: tests/h/activity/query_test.py [hunk 4]
```diff
@@ -175,20 +175,16 @@ def test_preserves_other_query_terms_for_user_search(
...
         unparse.assert_called_once_with({"tag": "foo"})
 
-    def test_does_nothing_with_non_matching_queries(self, pyramid_request, unparse):
+    def test_doesnt_raise_with_non_matching_queries(self, pyramid_request, unparse):
         query = MultiDict({"tag": "foo"})
 
-        result = check_url(pyramid_request, query, unparse=unparse)
-
-        assert result is None
+        check_url(pyramid_request, query, unparse=unparse)
```
[the remark this answers is not in the document]
Comment (marcospri) [reply tests/h/activity/query_test.py:181]: Oh, this one deserved a comment here. 

Pylint noticed that check_url always returns None (or raises) so it complained about assigning None. I'll add the assert but without a new assignment.

File: tests/h/assets_test.py [hunk 5]
```diff
@@ -77,7 +77,7 @@ def _fake_open(path):
 
         return None
 
-    open.side_effect = _fake_open
+    open_patch.side_effect = _fake_open
```
[the remark this answers is not in the document]
Comment (marcospri) [reply tests/h/assets_test.py:80]: I was feeling extra fancy, but this file has been removed from master

File: tests/h/services/user_signup_test.py [hunk 6]
```diff
@@ -155,7 +155,9 @@ def test_signup_logs_conflict_error_when_account_with_email_already_exists(
     def test_signup_raises_conflict_error_when_account_already_exists(
         self, svc, db_session, username, email
     ):
-        """This happens when two or more identical concurrent signup requests race each other to the db."""
+        """This happens when two or more identical concurrent signup
+        requests race each other to the db.
+        """
```
[the remark this answers is not in the document]
Comment (marcospri) [reply tests/h/services/user_signup_test.py:160]: pycodestyle is not yet running (only runs after a successful pylint run)

File: tests/h/activity/query_test.py (hunk 2 above)
Comment (marcospri) [reply tests/h/activity/query_test.py]: Same as before, pylint complained about ` assignment-from-none`, kept the assert now without an assignment

File: tests/h/assets_test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jon-betts) [reply tests/h/assets_test.py]: Total win

**Review by jon-betts — APPROVED** [review]
A lot of pretty boring work to do here, but it's good stuff!

**Pull request merged**